oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/userinfo/DefaultOAuth2UserService.java

```diff
@@ -64,6 +66,16 @@ public OAuth2User loadUser(OAuth2UserRequest userRequest) throws OAuth2Authentic
 			);
 			throw new OAuth2AuthenticationException(oauth2Error, oauth2Error.toString());
 		}
+		UserInfoRequestSchema userInfoRequestSchema = userRequest.getClientRegistration().getProviderDetails().getUserInfoEndpoint().getUserInfoRequestSchema();
```

Comment:
If the user has not configured `authenticationMethod` we should default to `HEADER`. So this check would not be needed.